### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by IYermakov**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -57,111 +56,202 @@ spec:
...
-                when { changeRequest target: 'master' }
-                    steps {
-                        container('docker') {
+        environment {
+            GREETING="${IMAGE_TAG}"
+        }
+        steps {
+            container('docker') {
+                script {
+                    IMAGE_ID = sh (script: 'docker build . -q --build-arg GREETING', returnStdout: true).trim()
```
Comment (tpolekhin) [inline Jenkinsfile:90]: have you tested this?
I think it should be double quotes `"$GREETING"` to interpolate variable

File: Jenkinsfile [hunk 2]
```diff
@@ -57,111 +56,202 @@ spec:
...
         }
     }
 
-    stage('Deploy to k8s') {
+    stage('Docker tag image and push to repository.') {
+        when { not { changeRequest() } }
+        steps {
+            container('docker') {
+                script {
+                    IMAGE_NAME = ("${GIT_BRANCH}"=='master') ? "${DOCKERHUB_REPO}/${HELM_RELEASE}" : "${DOCKERHUB_REPO}/${HELM_RELEASE}-${GIT_BRANCH}"
```
Comment (tpolekhin) [inline Jenkinsfile]: This isn't correct
docker image name shouldn't change no matter what branch we're building

**Review by tpolekhin — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (IYermakov) [reply Jenkinsfile]: I've thought you implied this as one of the first requirment. 
OK, I'll move it into tag.

File: Jenkinsfile (hunk 1 above)
Comment (IYermakov) [reply Jenkinsfile:90]: Yes. It works.

File: Jenkinsfile (hunk 2 above)
Comment (IYermakov) [reply Jenkinsfile]: Done

**Pull request merged**